Library/DiscUtils.Ext/ExtentsFileBuffer.cs

```diff
@@ -124,69 +134,91 @@ public override IEnumerable<StreamExtent> GetExtentsInRange(long start, long cou
                 new StreamExtent(start, count));
         }
 
+        /// <summary>
+        /// Returns the first extent containing a logical block greater than or equal to the given one.
+        /// Returns null if no such extent exists.
+        /// </summary>
         private Extent FindExtent(ExtentBlock node, uint logicalBlock)
         {
-            if (node.Index != null)
+            if (node.Extents != null)
+            {
+                return FindExtent(node.Extents, logicalBlock);
+            }
+            else if (node.Index != null)
+            {
+                return FindExtent(node.Index, logicalBlock);
+            }
+            else
             {
-                ExtentIndex idxEntry = null;
+                return null;
+            }
+        }
 
-                if (node.Index.Length == 0)
-                {
-                    return null;
-                }
-                if (node.Index[0].FirstLogicalBlock >= logicalBlock)
-                {
-                    idxEntry = node.Index[0];
-                }
-                else
+        /// <summary>
+        /// Returns the first extent containing a logical block greater than or equal to the given one.
+        /// Returns null if no such extent exists.
+        /// </summary>
+        private Extent FindExtent(Extent[] extents, uint logicalBlock)
+        {
+            for (int i = 0; i < extents.Length; ++i)
+            {
+                Extent extent = extents[i];
+                if (extent.FirstLogicalBlock >= logicalBlock ||
+                    extent.FirstLogicalBlock + extent.NumBlocks > logicalBlock)
                 {
-                    for (int i = 0; i < node.Index.Length; ++i)
-                    {
-                        if (node.Index[i].FirstLogicalBlock > logicalBlock)
-                        {
-                            idxEntry = node.Index[i - 1];
-                            break;
-                        }
-                    }
+                    return extent;
                 }
+            }
 
-                if (idxEntry == null)
-                {
-                    idxEntry = node.Index[node.Index.Length - 1];
-                }
+            return null;
+        }
 
-                ExtentBlock subBlock = LoadExtentBlock(idxEntry);
-                return FindExtent(subBlock, logicalBlock);
-            }
-            if (node.Extents != null)
+        /// <summary>
+        /// Returns the first extent containing a logical block greater than or equal to the given one.
+        /// Returns null if no such extent exists.
+        /// </summary>
+        private Extent FindExtent(ExtentIndex[] indexes, uint logicalBlock)
```

Comment:
Again, confirming this is just refactoring, right?

Reply:
Again, while the signature of this method is unchanged, the semantics have changed quite a bit.   It's more than simple refactoring.